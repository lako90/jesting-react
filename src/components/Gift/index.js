import React, { Component } from 'react';
import {
  Form,
  FormGroup,
  FormControl,
  Panel,
} from 'react-bootstrap';

class Gift extends Component {
  constructor() {
    super();

    this.state = {
      person: '',
      present: '',
    };
  }

  changePerson = () => {
    this.setState({ person: this.inputPerson.value });
  }

  changePresent = () => {
    this.setState({ present: this.inputPresent.current.value });
  }

  render() {
    return (
      <Panel>
        <Form>
          <FormGroup>
            <FormControl
              // type={'text'}
              // placeholder={'Person'}
              className={'input-person'}
              // inputRef={(ref) => { this.inputPerson = ref }}
              // onChange={this.changePerson}
            />
            {/* <FormControl
              type={'text'}
              placeholder={'Present'}
              className={'input-present'}
              inputRef={this.inputPresent}
              onChange={this.changePresent}
            /> */}
          </FormGroup>
        </Form>
      </Panel>
    );
  }
}

export default Gift;
