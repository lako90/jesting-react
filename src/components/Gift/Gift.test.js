import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Gift from '.';

Enzyme.configure({ adapter: new Adapter() });

describe('Gift', () => {
  const gift = shallow(<Gift />);

  it('renders correctly', () => {
    expect(gift).toMatchSnapshot();
  })

  it('initializes the `state` with an empty person and present', () => {
    expect(gift.state()).toEqual({ person: '', present: '' });
  })

  describe('when typing into `person` input', () => {
    const person = "ExamplePerson";

    beforeAll(() => {
      gift.find('.input-person').simulate('change', { target: { value: person } })
    });

    it('change corretly the `state`', () => {
      expect(gift.state().person).toEqual(person);
    });

    // it('change corretly the `input`', () => {
    //   expect(gift.find('.input-person').value).toEqual(person)
    // })
  });

  describe('when typing into `present` input', () => {
    const present = "ExamplePresent";

    beforeEach(() => {
      gift.find('.input-present').simulate('change', { target: { value: present } })
    });

    it('change corretly the `state`', () => {
      expect(gift.state().present).toEqual(present);
    });

    // it('change corretly the `input`', () => {
    //   expect(gift.find('.input-present').value).toEqual(present)
    // })
  });
})
