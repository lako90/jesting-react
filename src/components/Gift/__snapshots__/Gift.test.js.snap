// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Form
        bsClass="form"
        componentClass="form"
        horizontal={false}
        inline={false}
>
        <FormGroup
                bsClass="form-group"
        >
                <FormControl
                        bsClass="form-control"
                        className="input-person"
                        componentClass="input"
                />
        </FormGroup>
</Form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "form",
        "children": <FormGroup
          bsClass="form-group"
>
          <FormControl
                    bsClass="form-control"
                    className="input-person"
                    componentClass="input"
          />
</FormGroup>,
        "componentClass": "form",
        "horizontal": false,
        "inline": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": <FormControl
            bsClass="form-control"
            className="input-person"
            componentClass="input"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-control",
            "className": "input-person",
            "componentClass": "input",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
>
          <FormGroup
                    bsClass="form-group"
          >
                    <FormControl
                              bsClass="form-control"
                              className="input-person"
                              componentClass="input"
                    />
          </FormGroup>
</Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": <FormGroup
            bsClass="form-group"
>
            <FormControl
                        bsClass="form-control"
                        className="input-person"
                        componentClass="input"
            />
</FormGroup>,
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <FormControl
              bsClass="form-control"
              className="input-person"
              componentClass="input"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "className": "input-person",
              "componentClass": "input",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
